packages/itmat-ui-react/src/App.tsx

```diff
@@ -1,36 +1,42 @@
 import * as React from 'react';
 import { ApolloProvider, Query } from 'react-apollo';
-import { BrowserRouter as Router } from 'react-router-dom';
+import { BrowserRouter as Router, Switch, Route, RouteComponentProps } from 'react-router-dom';
 import { client } from './components/apolloClient';
 import { LoginBox } from './components/login/login';
 import { Spinner } from './components/reusable/icons/spinner';
 import { MainMenuBar } from './components/scaffold/mainMenuBar';
 import { MainPanel } from './components/scaffold/mainPanel';
 import css from './components/scaffold/scaffold.module.css';
+import { ResetPasswordPage } from './components/resetPasswordPage/resetPasswordPage';
+import { RequestResetPassword } from './components/resetPasswordPage/requestResetPasswordPage';
 import { WHO_AM_I } from 'itmat-commons/dist/graphql/user';
 
 class App extends React.Component {
     public render() {
         return (
             <ApolloProvider client={client}>
-                <Query<any, any> query={WHO_AM_I}>
-                    {({ loading, error, data }) => {
-                        if (loading) { return <div style={{ width: '100%', height: '100%', textAlign: 'center', paddingTop: '20%' }}><Spinner /></div>; }
-                        if (error) { return <p>Error :( {error.message}</p>; }
-                        if (data.whoAmI !== null && data.whoAmI !== undefined && data.whoAmI.username !== null) { // if logged in return the app
-                            // return <div className={css.app}>
-                            return <div className={css.app + ' dark_theme'}>
-                                <Router>
-                                    <>
-                                        <MainMenuBar projects={data.whoAmI.access.projects} />
-                                        <MainPanel />
-                                    </>
-                                </Router>
-                            </div>;
-                        }
-                        return <LoginBox />; // if not logged in return the login boxs
-                    }}
-                </Query>
+                <Router>
+                    <Switch>
+                        <Route path='/resetPassword/:username/:token' render={({ match }: RouteComponentProps<{ username: string, token: string }>) => <ResetPasswordPage matchparams={match.params}/>}/>
```

Comment:
The type declaration should be done in the component file.

Reply:
what do you mean?

Reply:
The `RouteComponentProps<{}>` should be placed in the `resetPasswordPage.tsx`. Also, in this context, I am not sure why you are using `<Route render={} />` instead of `<Route component={} />` since you are not really making use of an inline function renderer.
To a larger extend, [the recommended usage](https://reacttraining.com/react-router/web/api/Route/route-render-methods) structure is to provide a child to `<Route />` and utilise `useRouterMatch()` in the child component.